servers/src/grin/seed.rs

```diff
@@ -315,7 +315,7 @@ fn listen_for_addrs(
 	let addrs: Vec<SocketAddr> = rx.try_iter().collect();
 
 	// If we have a healthy number of outbound peers then we are done here.
-	if peers.healthy_peers_mix() {
+	if peers.peer_count() > peers.peer_outbound_count() && peers.healthy_peers_mix() {
```

Comment:
So we're basically saying we do not need to try initiating more outbound connections if - 
* _at least_ one inbound connection, and
* healthy_peers_mix

And `healthy_peers_mix` is - 

* peer_count >= min_preferred, and
* outbound_count >= min_preferred / 2

And that first check (at least one inbound connection) is basically to determine if we are behind a NAT or not?

I think I get the logic but it feel confusing.

Is it worth introducing a `peer.peer_inbound_count()` fn to make this more a little more explicit?
And then branch clearly on that first check - "do this if we think we are a public node, otherwise do this other thing"?